src/v3/src/util/appendLoginHint.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+export const appendLoginHint = (deviceChallengeUrl = '', loginHint = ''): string => {
+  if (deviceChallengeUrl && loginHint) {
+    return `${deviceChallengeUrl}&login_hint=${loginHint}`;
+  }
+  return deviceChallengeUrl;
+};
```

Comment:
Agree on both counts, we can rename the fn and also improve on the original implementation in v2 by using `URL` and `URLSearchParams` while we're at it

`URL`: https://developer.mozilla.org/en-US/docs/Web/API/URL/URL
`URL.searchParams` is a `URLSearchParams` object: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
I found a problem in that this breaks the spec test since the spec test uses `okta-verify.html` as the dummy url.  The `URL` object constructor does not accept that as a valid url and throws an error:
<img width="734" alt="image" src="https://user-images.githubusercontent.com/107433508/214707384-dd6d0415-82a3-4815-a38e-e2214005735c.png">

How do we want to handle this do we want to change the test mock file?